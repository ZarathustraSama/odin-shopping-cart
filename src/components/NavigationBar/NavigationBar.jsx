import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import './NavigationBar.module.css';

function NavigationBar({ shoppingCart = [], checkout = false }) {

  function countItems(shoppingItems) {
    let count = 0;

    for (let i = 0; i < shoppingItems.length; i++) {
      count += shoppingItems[i].quantity;
    }

    return count;
  }

  return (
    <nav>
      <Link to="/">Home</Link>
      <Link to="/shop">Shop</Link>
      <span>Your cart currently has {countItems(shoppingCart)} items. </span>
      {checkout ? <button className="checkoutBtn">Checkout</button> : ""}
    </nav>
  )
}

NavigationBar.propTypes = {
  shoppingCart: PropTypes.array.isRequired,
  checkout: PropTypes.bool.isRequired,
}

export default NavigationBar;