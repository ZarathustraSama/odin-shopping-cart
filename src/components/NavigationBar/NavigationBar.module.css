.a {
  border: 1px solid black;
  color: black;
  font-size: large;
  font-weight: bolder;
  padding: 1em;
  margin: 1em;
  text-decoration: none;
}

.a:hover {
  background-color: gray;
}

.button, .input {
  background-color: white;
  border: 3px solid gold;
  font-size: large;
}

.button:hover, .input:hover {
  background-color: aquamarine;
}

.nav {
  padding: 1em;
  text-align: center;
}

.span {
  font-size: large;
}