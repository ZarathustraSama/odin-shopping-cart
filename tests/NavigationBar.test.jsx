import { describe, it, expect } from 'vitest';
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import NavigationBar from '../src/components/NavigationBar/NavigationBar';

describe("NavigationBar component", () => {
  it("renders a simple navbar with links and a span element", () => {
    const { container } = render(<BrowserRouter><NavigationBar /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  })

  it("does not render a button when in the home route", () => {
    render(<BrowserRouter><NavigationBar /></BrowserRouter>);
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
  })

  it("does render a button when in the shop route", () => {
    render(<BrowserRouter><NavigationBar checkout={true} /></BrowserRouter>);
    expect(screen.getByRole("button")).toBeInTheDocument();
  })

  it("counts correctly the number of items in the cart", () => {
    const shoppingCart = [
                          {name: "item1", quantity: 5},
                          {name: "item2", quantity: 6},
                          {name: "item3", quantity: 4}
                        ];
    
    render(<BrowserRouter><NavigationBar shoppingCart={shoppingCart} /></BrowserRouter>);
    expect(screen.getByRole("textbox").textContent).toMatch(/Your cart currently has 15 items./);  
  })
})