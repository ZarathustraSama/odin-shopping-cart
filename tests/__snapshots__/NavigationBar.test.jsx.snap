// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBar component > renders a simple navbar with links and a span element 1`] = `
<div>
  <nav>
    <a
      data-discover="true"
      href="/"
    >
      Home
    </a>
    <a
      data-discover="true"
      href="/shop"
    >
      Shop
    </a>
    <span
      role="textbox"
    >
      Your cart currently has 
      0
       items. 
    </span>
  </nav>
</div>
`;
